modules/thunder_article/src/ThunderNodeForm.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Drupal\thunder_article;
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\node\NodeForm;
+
+/**
+ * Base for handler for node edit forms.
+ */
+class ThunderNodeForm extends NodeForm {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function actions(array $form, FormStateInterface $form_state) {
+    $element = parent::actions($form, $form_state);
+
+    if ($element['publish']['#access'] && \Drupal::currentUser()->hasPermission('administer nodes')) {
```

Comment:
This should be more strict checking if "$element['publish']" doesn't exist. Also for:
`min($element['publish']['#weight'], $element['unpublish']['#weight'])`.